### PR TITLE
feat: made chat component

### PR DESCRIPTION
**Pull request by RaznoOleg**
## Description changes:
• Made chat component;
• Made the input change size when moving to a new line;
• Made the user's messages appear on the right and the other user's messages on the left;
• Made it show a message that another user is typing;
• Made it possible to send messages by pressing "Enter";
• Made it possible to enter a maximum of 250 messages in the input;
• Made so that if the message is large, it is automatically transferred to a new line;
• Added automatic scrolling down if a new message is sent;
• Added that the user cannot send an empty message.

### General
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Don't have "any" on my code
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] All links in on ENV files
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

#### Overall view of the chat and exchange messages in chat:
image

#### Transition to a new line with the "Enter + Shift" key combination:
image
image

#### Sending a large message and automatic transfer to a new line:
image
image

#### Display in real time if the user types a message:
image
image

### CODE REVIEW

File: src/components/Chat/index.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+function Chat() {
+  const { t } = useTranslation();
+
+  const doctorData = useAppSelector((state) => state.doctorReducer);
+
+  const doctorId = doctorData.id;
+
+  const [socket, setSocket] = useState<Socket>();
+
```
Comment (lyudad) [inline src/components/Chat/index.tsx]: this empty line not needed

File: src/components/Chat/index.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  const [messages, setMessages] = useState<Message[]>([]);
+
```
Comment (lyudad) [inline src/components/Chat/index.tsx]: this empty line not needed

File: src/components/Chat/index.tsx (hunk 1 above)
Comment (RaznoOleg) [reply src/components/Chat/index.tsx]: fixed

File: src/components/Chat/index.tsx (hunk 2 above)
Comment (RaznoOleg) [reply src/components/Chat/index.tsx]: fixed

**Pull request merged**